File: server/src/main/kotlin/com/pixapencil/server/domain/Creation.kt
package com.pixapencil.server.domain

import jakarta.persistence.*
import org.hibernate.annotations.CreationTimestamp
import java.time.LocalDateTime

@Table(name = "creations")
@Entity
class Creation(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @Column(nullable = false)
    var title: String,

    @Column(nullable = false, length = 5000)
    var description: String,

    @Column(nullable = false)
    var imageKey: String,

    @Column(nullable = false)
    var likeCount: Int = 0,

    @CreationTimestamp
    var createdAt: LocalDateTime = LocalDateTime.now(),

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    var user: User,

    @ManyToMany(mappedBy = "likedCreations")
    val likedBy: MutableList<User> = mutableListOf(),

    @Column(nullable = false)
    var commentCount: Int = 0,
)

File: web/src/models/Creation/Creation.ts
import { Author } from "../Author/Author"

export interface Creation {
  id: number,
  title: string
  description: string
  imageUrl: string
  isLiked: boolean
  likeCount: number
  author: Author
  createdAt: string
}

export interface UploadCreation {
  title: string
  description: string
  file?: File
}
